modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep_chunks/aah_sweep_chunk.cc

```diff
@@ -39,277 +37,242 @@ AAH_SweepChunk::Sweep(AngleSet& angle_set)
   int preloc_face_counter = -1;
 
-  sweep_dependency_interface_.angle_set_ = &angle_set;
-  sweep_dependency_interface_.surface_source_active_ = IsSurfaceSourceActive();
-  sweep_dependency_interface_.gs_ss_begin_ = gs_ss_begin_;
-  sweep_dependency_interface_.gs_gi_ = gs_gi_;
+  auto& fluds = dynamic_cast<AAH_FLUDS&>(angle_set.GetFLUDS());
+  const auto& m2d_op = groupset_.quadrature_->GetMomentToDiscreteOperator();
+  const auto& d2m_op = groupset_.quadrature_->GetDiscreteToMomentOperator();
 
-  auto& aah_sweep_depinterf =
-    dynamic_cast<AAH_SweepDependencyInterface&>(sweep_dependency_interface_);
-  aah_sweep_depinterf.fluds_ = &dynamic_cast<AAH_FLUDS&>(angle_set.GetFLUDS());
+  std::vector<std::vector<double>> Amat(max_num_cell_dofs_,
+                                        std::vector<double>(max_num_cell_dofs_));
+  std::vector<std::vector<double>> Atemp(max_num_cell_dofs_,
+                                         std::vector<double>(max_num_cell_dofs_));
+  std::vector<std::vector<double>> b(groupset_.groups_.size(),
+                                     std::vector<double>(max_num_cell_dofs_));
+  std::vector<double> source(max_num_cell_dofs_);
 
   // Loop over each cell
   const auto& spds = angle_set.GetSPDS();
   const auto& spls = spds.GetSPLS().item_id;
   const size_t num_spls = spls.size();
   for (size_t spls_index = 0; spls_index < num_spls; ++spls_index)
   {
-    cell_local_id_ = spls[spls_index];
-    cell_ = &grid_.local_cells[cell_local_id_];
-    sweep_dependency_interface_.cell_ptr_ = cell_;
-    sweep_dependency_interface_.cell_local_id_ = cell_local_id_;
-    cell_mapping_ = &grid_fe_view_.GetCellMapping(*cell_);
-    cell_transport_view_ = &grid_transport_view_[cell_->local_id_];
-
-    const auto& face_orientations = spds.CellFaceOrientations()[cell_local_id_];
+    auto cell_local_id = spls[spls_index];
+    auto& cell = grid_.local_cells[cell_local_id];
+    auto& cell_mapping = grid_fe_view_.GetCellMapping(cell);
+    auto& cell_transport_view = grid_transport_view_[cell_local_id];
+    auto cell_num_faces = cell.faces_.size();
+    auto cell_num_nodes = cell_mapping.NumNodes();
 
-    cell_num_faces_ = cell_->faces_.size();
-    cell_num_nodes_ = cell_mapping_->NumNodes();
-    const auto& sigma_t = xs_.at(cell_->material_id_)->SigmaTotal();
+    const auto& face_orientations = spds.CellFaceOrientations()[cell_local_id];
+    std::vector<double> face_mu_values(cell_num_faces);
 
-    aah_sweep_depinterf.spls_index = spls_index;
+    const auto& sigma_t = xs_.at(cell.material_id_)->SigmaTotal();
 
-    // Get Cell matrices
-    const auto& fe_intgrl_values = unit_cell_matrices_[cell_local_id_];
-    G_ = &fe_intgrl_values.intV_shapeI_gradshapeJ;
-    M_ = &fe_intgrl_values.intV_shapeI_shapeJ;
-    M_surf_ = &fe_intgrl_values.intS_shapeI_shapeJ;
-    IntS_shapeI_ = &fe_intgrl_values.intS_shapeI;
+    // Get cell matrices
+    const auto& G = unit_cell_matrices_[cell_local_id].intV_shapeI_gradshapeJ;
```

Comment:
that must be the PDT way, with 2 integration by parts, where only 1 applies upwinding. otherwise, I would have expected the name to be intV_shapeJ_gradshapeI. So, I suppose the surface mass matrix M_surf ONLY applied to the incoming faces.